java/src/com/ibm/streamsx/topology/builder/BOperatorInvocation.java

```diff
@@ -123,28 +118,40 @@ public void setParameter(String name, Object value) {
 
         // Set the value in the OperatorInvocation.
         
-        if (value instanceof UString) {
-            value = ((UString) value).value();
-            jsonValue = value;
-            jsonType = "ustring";
-        } else if (value instanceof UnsignedByte) {
-            value = ((UnsignedByte) value).value();
-            jsonValue = value;
-            jsonType = "unsigned";
-        } else if (value instanceof UnsignedShort) {
-            value = ((UnsignedShort) value).value();
-            jsonValue = value;
-            jsonType = "unsigned";
-        } else if (value instanceof UnsignedInteger) {
-            value = ((UnsignedInteger) value).value();
-            jsonValue = value;
-            jsonType = "unsigned";
-        } else if (value instanceof UnsignedLong) {
-            value = ((UnsignedLong) value).value();
-            jsonValue = value;
-            jsonType = "unsigned";
+        if (value instanceof JSONAble) {
+            value = ((JSONAble)value).toJSON();
         }
-        
+        if (value instanceof JSONObject) {
+            JSONObject jo = ((JSONObject) value);
+            if (jo.get("type") == null || jo.get("value") == null)
+                throw new IllegalArgumentException("Illegal JSONObject " + jo);
+            String type = (String) jo.get("type");
+            Object val = (JSONObject) jo.get("value");
+            if ("__spl_wrappedValue".equals(type)) {
```

Comment:
Seems like this is just an SPL value, not an "SPL wrapped value", maybe `__spl_value` as the type identifier?